movie_search/tmdb_api.py

```diff
@@ -17,16 +18,39 @@ def __init__(self, language = None):
         self.language = language or settings.TMDB_API_LANG
 
     def get_data_from_endpoint(self, endpoint, **kwargs):
-        url = f"{self.base_url}{endpoint}"
-        params = {"api_key": self.api_key, "language": self.language}
-        params.update(kwargs)
-        response = requests.get(url, params=params)
-        response.raise_for_status()
-        return response.json()
+            url = f"{self.base_url}{endpoint}"
+            params = {"api_key": self.api_key, "language": self.language}
+            params.update(kwargs)
+            response = requests.get(url, params=params)
+
+            # Attempt to parse the response as JSON
+            try:
+                return response.json()
+            except JSONDecodeError:
+                # Handle JSON decoding error
+                print(f"Failed to parse JSON response for URL: {url}")
+                print(f"HTTP Status Code: {response.status_code}")
+                print("Response Text:", response.text[:500])  # Print first 500 characters of the response
+                error_message = f"Failed to parse JSON response for URL: {url} - Status Code: {response.status_code} - Response: {response.text[:500]}"
+                print(error_message)
+                # Optionally, raise an exception or return a default value
+                raise JSONDecodeError(error_message)
 
     def get_data_by_query(self, endpoint, text_query, result_key):
         response_json = self.get_data_from_endpoint(endpoint, query=text_query)
         data = response_json["results"]
         return {row[result_key]: row["id"] for row in data}
+    
+
+    def main():
+        from utils import dump_movie_data_to_json
```

Comment:
any reason to do the import here? ideally imports should only be done at the top of the module